File: fastlogging/examples/file_add_writer.rs
use std::path::PathBuf;

use fastlogging::{CompressionMethodEnum, DEBUG, FileWriterConfig, Logging, LoggingError};

fn main() -> Result<(), LoggingError> {
    let mut logger = Logging::new(DEBUG, "root", None, None, None)?;
    logger.add_writer_config(
        &FileWriterConfig::new(
            DEBUG,
            PathBuf::from("/tmp/cfastlogging.log"),
            1024,
            3,
            None,
            None,
            Some(CompressionMethodEnum::Store),
        )?
        .into(),
    )?;
    logger.trace("Trace Message")?;
    logger.debug("Debug Message")?;
    logger.info("Info Message")?;
    logger.success("Success Message")?;
    logger.error("Error Message")?;
    logger.fatal("Fatal Message")?;
    logger.shutdown(false)?;
    Ok(())
}

File: cxxfastlogging/examples/file_add_writer.cpp
#include "cxxfastlogging/h/fastlogging.h"

int main()
{
    auto log = Logging::new_default();
    log->add_writer_config(WriterConfig::new_file(
        DEBUG, "cxxfastlogging.log", 1024, 3, -1, -1,
        CompressionMethodEnum::Store));
    log->trace("Trace Message");
    log->debug("Debug Message");
    log->info("Info Message");
    log->success("Success Message");
    log->warning("Warning Message");
    log->error("Error Message");
    log->fatal("Fatal Message");
    log->shutdown(false);
    return 0;
}
